### ISSUE TITLE
[Feature Request] Ability to control what characters are "striped" from suggestions

### ISSUE THREAD

**Issue by einirpals** [tadashi-aikawa/obsidian-various-complements-plugin#170]
It would be nice to be able to control what characters are striped from the suggestions to support more languages. The plugin already strips the English straight single ( ' ) and double quotation ( " ) marks from the suggestions.

For example, in Icelandic the quotation mark convention (rule) is like this „…“ and ‚…‘ but these characters are not removed from the query so the suggestions are always empty.

![Screenshot 2022-10-08 at 09 26 49](https://user-images.githubusercontent.com/15929776/194700536-f2986fe7-36af-401f-baae-9f76b556e42c.jpg)

In Icelandic this is not only good typography, but the right way to do quotes, and given that I do some academic work in obsidian, I run in to this frequently.

Anyway love this plugin, keep up the good work.

**Comment by tadashi-aikawa** [comment]
Hi, @einirpals. Sorry for my late answer.

I have never known that there are many styles for quations. I'll consider the feature 👍

**Comment by tadashi-aikawa** [comment]
@einirpals 
I have checked the behavior in response to this issue, and it appears that treating ",," as double quotes does not interfere with the behavior. If you don't mind, could you please tell us how to reproduce the case where the problem occurs in the sandbox?

**Comment by einirpals** [comment]
@tadashi-aikawa 
Sure, no problem. I will post a small video of the behavior.

https://github.com/tadashi-aikawa/obsidian-various-complements-plugin/assets/15929776/4a6ceaa7-e9b6-4b22-8bc7-416df2e577eb

And the Icelandic quotation mark is this character -> [Wiki](https://en.wiktionary.org/wiki/%E2%80%9E)

**Comment by tadashi-aikawa** [comment]
@einirpals 
Thank you!

Could you tell me your settings copied by the [Copy plugin settings](https://tadashi-aikawa.github.io/docs-obsidian-various-complements-plugin/3.%20Commands/%F0%9F%8E%AE%20Copy%20plugin%20settings/) command?

**Comment by einirpals** [comment]
@tadashi-aikawa 
Here you go.

{
    "version": "7.3.0",
    "mobile": false,
    "settings": {
        "strategy": "default",
        "matchStrategy": "prefix",
        "maxNumberOfSuggestions": 20,
        "maxNumberOfWordsAsPhrase": 1,
        "minNumberOfCharactersTriggered": 2,
        "minNumberOfWordsTriggeredPhrase": 1,
        "complementAutomatically": true,
        "delayMilliSeconds": 170,
        "disableSuggestionsDuringImeOn": false,
        "insertAfterCompletion": false,
        "firstCharactersDisableSuggestions": ":/^",
        "showMatchStrategy": false,
        "showComplementAutomatically": false,
        "showIndexingStatus": false,
        "descriptionOnSuggestion": "Full",
        "selectSuggestionKeys": "Tab",
        "additionalCycleThroughSuggestionsKeys": "None",
        "disableUpDownKeysForCycleThroughSuggestionsKeys": false,
        "openSourceFileKey": "None",
        "propagateEsc": false,
        "enableCurrentFileComplement": true,
        "currentFileMinNumberOfCharacters": 0,
        "onlyComplementEnglishOnCurrentFileComplement": false,
        "enableCurrentVaultComplement": false,
        "currentVaultMinNumberOfCharacters": 0,
        "includeCurrentVaultPathPrefixPatterns": "",
        "excludeCurrentVaultPathPrefixPatterns": "",
        "includeCurrentVaultOnlyFilesUnderCurrentDirectory": false,
        "enableCustomDictionaryComplement": true,
        "customDictionaryPaths": ".VCDictionary/IS_A_VariousComplementsDictionary.json\n.VCDictionary/IS_B_VariousComplementsDictionary.json\n.VCDictionary/IS_C_VariousComplementsDictionary.json\n.VCDictionary/IS_D_VariousComplementsDictionary.json\n.VCDictionary/IS_E_VariousComplementsDictionary.json\n.VCDictionary/IS_F_VariousComplementsDictionary.json\n.VCDictionary/IS_G_VariousComplementsDictionary.json\n.VCDictionary/IS_H_VariousComplementsDictionary.json\n.VCDictionary/IS_I_VariousComplementsDictionary.json\n.VCDictionary/IS_J_VariousComplementsDictionary.json\n.VCDictionary/IS_K_VariousComplementsDictionary.json\n.VCDictionary/IS_L_VariousComplementsDictionary.json\n.VCDictionary/IS_M_VariousComplementsDictionary.json\n.VCDictionary/IS_N_VariousComplementsDictionary.json\n.VCDictionary/IS_O_VariousComplementsDictionary.json\n.VCDictionary/IS_P_VariousComplementsDictionary.json\n.VCDictionary/IS_Q_VariousComplementsDictionary.json\n.VCDictionary/IS_R_VariousComplementsDictionary.json\n.VCDictionary/IS_S_VariousComplementsDictionary.json\n.VCDictionary/IS_T_VariousComplementsDictionary.json\n.VCDictionary/IS_U_VariousComplementsDictionary.json\n.VCDictionary/IS_V_VariousComplementsDictionary.json\n.VCDictionary/IS_W_VariousComplementsDictionary.json\n.VCDictionary/IS_X_VariousComplementsDictionary.json\n.VCDictionary/IS_Y_VariousComplementsDictionary.json\n.VCDictionary/IS_Z_VariousComplementsDictionary.json\n.VCDictionary/IS_Á_VariousComplementsDictionary.json\n.VCDictionary/IS_Æ_VariousComplementsDictionary.json\n.VCDictionary/IS_É_VariousComplementsDictionary.json\n.VCDictionary/IS_Í_VariousComplementsDictionary.json\n.VCDictionary/IS_Ó_VariousComplementsDictionary.json\n.VCDictionary/IS_Ö_VariousComplementsDictionary.json\n.VCDictionary/IS_Ú_VariousComplementsDictionary.json\n.VCDictionary/IS_Ý_VariousComplementsDictionary.json\n.VCDictionary/IS_Þ_VariousComplementsDictionary.json",
        "columnDelimiter": "Tab",
        "customDictionaryWordRegexPattern": "",
        "delimiterToHideSuggestion": "",
        "delimiterToDivideSuggestionsForDisplayFromInsertion": "",
        "caretLocationSymbolAfterComplement": "",
        "displayedTextSuffix": " => ...",
        "enableInternalLinkComplement": true,
        "suggestInternalLinkWithAlias": false,
        "excludeInternalLinkPathPrefixPatterns": "",
        "enableFrontMatterComplement": true,
        "frontMatterComplementMatchStrategy": "inherit",
        "insertCommaAfterFrontMatterCompletion": false,
        "intelligentSuggestionPrioritization": {
            "maxDaysToKeepHistory": 30,
            "maxNumberOfHistoryToKeep": 0
        },
        "showLogAboutPerformanceInConsole": true,
        "selectionHistoryTree": null
    }
}

**Comment by tadashi-aikawa** [comment]
@einirpals
Thank you! I could reproduce it👍

Sorry, I mistook „Halló“ for ,,Halló'' 😅

**Comment by tadashi-aikawa** [comment]
Wow.. There are many quotation marks and patterns! I have added them as far as I know 👍 

```
describe.each<{
  content: string;
  expected: { word: string; offset: number }[];
}>`
  content                  | expected
  ${'"aaa" "bbb"'}         | ${["aaa", "bbb"]}
  ${"'aaa' 'bbb'"}         | ${["aaa", "bbb"]}
  ${"„aaa“ „bbb“"}         | ${["aaa", "bbb"]}
  ${"‚aaa’ ‚bbb’"}         | ${["aaa", "bbb"]}
  ${"“aaa” “bbb”"}         | ${["aaa", "bbb"]}
  ${"‘aaa’ ‘bbb’"}         | ${["aaa", "bbb"]}
  ${"„aaa” „bbb”"}         | ${["aaa", "bbb"]}
  ${"‚aaa’ ‚bbb’"}         | ${["aaa", "bbb"]}
  ${"«aaa» «bbb»"}         | ${["aaa", "bbb"]}
  ${"‹aaa› ‹bbb›"}         | ${["aaa", "bbb"]}
  ${"»aaa« »bbb«"}         | ${["aaa", "bbb"]}
  ${"›aaa‹ ›bbb‹"}         | ${["aaa", "bbb"]}
  ${"”aaa” ”bbb”"}         | ${["aaa", "bbb"]}
  ${"’aaa’ ’bbb’"}         | ${["aaa", "bbb"]}
  ${"»aaa» »bbb»"}         | ${["aaa", "bbb"]}
  ${"›aaa› ›bbb›"}         | ${["aaa", "bbb"]}
  ${"“aaa„ “bbb„"}         | ${["aaa", "bbb"]}
  ${"‘aaa‚ ‘bbb‚"}         | ${["aaa", "bbb"]}
  ${"‘aaa’ ‘bbb’"}         | ${["aaa", "bbb"]}
`("tokenize", ({ content, expected }) => {
  test(`tokenize(${content}) = ${expected}`, () => {
    expect(new DefaultTokenizer().tokenize(content)).toStrictEqual(expected);
  });
});
```

[references] (Written in Japanese)
https://www.weblio.jp/wkpja/content/%E5%BC%95%E7%94%A8%E7%AC%A6_%E4%B8%80%E8%A6%A7

**Comment by tadashi-aikawa** [comment]
LGTM. I'll release it as v9.0.0-beta2.

![2023-08-26-20-47-31](https://github.com/tadashi-aikawa/obsidian-various-complements-plugin/assets/9500018/c22b79a8-5d94-4cf0-a930-06602c2e1ba4)

**Comment by tadashi-aikawa** [comment]
@einirpals 
I have released it in [v9.0.0-beat2](https://github.com/tadashi-aikawa/obsidian-various-complements-plugin/releases/tag/9.0.0-beta2) 🚀 

If you use BRAT, I would appreciate it if you could check it out before the official release.

**Comment by einirpals** [comment]
@tadashi-aikawa 
Cool. I will check it out.
And thanks again for this plugin and checking out this little issue.
